src/Ursa/Controls/PathPicker/PathPicker.cs

```diff
@@ -0,0 +1,279 @@
+            nameof(DefaultFileExtension), string.Empty);
+
+    public static readonly DirectProperty<PathPicker, IReadOnlyList<string>> SelectedPathsProperty =
+        AvaloniaProperty.RegisterDirect<PathPicker, IReadOnlyList<string>>(
+            nameof(SelectedPaths), o => o.SelectedPaths, (o, v) => o.SelectedPaths = v);
+
+    public static readonly StyledProperty<ICommand?> CommandProperty = AvaloniaProperty.Register<PathPicker, ICommand?>(
+        nameof(Command));
+
+    public static readonly StyledProperty<bool> AllowMultipleProperty = AvaloniaProperty.Register<PathPicker, bool>(
+        nameof(AllowMultiple));
+
+    public static readonly StyledProperty<string?> SelectedPathsTextProperty =
+        AvaloniaProperty.Register<PathPicker, string?>(
+            nameof(SelectedPathsText), defaultBindingMode: BindingMode.TwoWay);
+
+    public string? SelectedPathsText
+    {
+        get => GetValue(SelectedPathsTextProperty);
+        set => SetValue(SelectedPathsTextProperty, value);
+    }
+
+    private Button? _button;
+
+    private IReadOnlyList<string> _selectedPaths = [];
+
+    public bool AllowMultiple
+    {
+        get => GetValue(AllowMultipleProperty);
+        set => SetValue(AllowMultipleProperty, value);
+    }
+
+    public ICommand? Command
+    {
+        get => GetValue(CommandProperty);
+        set => SetValue(CommandProperty, value);
+    }
+
+    public IReadOnlyList<string> SelectedPaths
+    {
+        get => _selectedPaths;
+        private set => SetAndRaise(SelectedPathsProperty, ref _selectedPaths, value);
+    }
+
+    public string SuggestedFileName
+    {
+        get => GetValue(SuggestedFileNameProperty);
+        set => SetValue(SuggestedFileNameProperty, value);
+    }
+
+    public string DefaultFileExtension
+    {
+        get => GetValue(DefaultFileExtensionProperty);
+        set => SetValue(DefaultFileExtensionProperty, value);
+    }
+
+
+    public string Title
+    {
+        get => GetValue(TitleProperty);
+        set => SetValue(TitleProperty, value);
+    }
+
+    public string FileFilter
+    {
+        get => GetValue(FileFilterProperty);
+        set => SetValue(FileFilterProperty, value);
+    }
+
+    public UsePickerTypes UsePickerType
+    {
+        get => GetValue(UsePickerTypeProperty);
+        set => SetValue(UsePickerTypeProperty, value);
+    }
+
+    public string SuggestedStartPath
+    {
+        get => GetValue(SuggestedStartPathProperty);
+        set => SetValue(SuggestedStartPathProperty, value);
+    }
+
+    private bool _twoConvertLock;
+
+    protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
+    {
+        base.OnPropertyChanged(change);
+        if (_twoConvertLock) return;
+        if (change.Property == SelectedPathsProperty)
+        {
+            _twoConvertLock = true;
+            var stringBuilder = new StringBuilder();
+            stringBuilder.Append(SelectedPaths.FirstOrDefault());
```

Comment:
Just use for iteration to avoid un-necessary multiple LINQ call and enumeration.